modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/CreateDumpFutureTask.java

```diff
@@ -141,7 +147,8 @@ public CreateDumpFutureTask(
         );
 
         this.dumpDir = dumpDir;
-        this.ioFactory = ioFactory;
+        this.ioFactory = compress ? new WriteOnlyZipFileIOFactory() : ioFactory;
```

Comment:
It means that we try to zip every entry? I suppose that compression block of the data (that contains at least some keys) should be much efficient. What do you think if compress the final file instead of every entry?